src/components/BackButton.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import Button from './Button';
+
+function BackButton(props) {
+    const { setElementToShow, setUserPoints, setCurrentPage } = props;
+    return (
```

Comment:
Czy nie lepiej stworzyć u rodzica jedną funkcje handleClick w której zmienisz stan i przekazać tylką ją do BackButton na onClicku?
Bo tu jest to trochę dziwne bo stan jest jakby przypisany do komponentu, używasz skłądni "set"Element skojarzoną ze zmianą stanu a tu nie ma w tym komponencie żadnego stanu. Stąd wydaje mi się(mogę sie mylić), że handleClick byłby chyba praktyczniejszy.
Ale to i tak bez znaczenia jak działa :D